python/pyspark/ml/torch/distributor.py

```diff
@@ -0,0 +1,297 @@
+    >>>     return trainer
+    >>> distributor = TorchDistributor(framework="pytorch-lightning",
+                                  num_processes=num_proc,
+                                  local_mode=True,
+                                  use_gpu=True)
+    >>> trainer = distributor.run(train)
+    """
+
+    available_frameworks = ["pytorch", "pytorch-lightning"]
+
+    def __init__(
+        self,
+        framework: str,
+        num_processes: int = 1,
+        local_mode: bool = True,
+        use_gpu: bool = True,
+        spark: Optional[SparkSession] = None,
+    ):
+        """Initializes the distributor.
+
+        Parameters
+        ----------
+        framework : str
+            A string indicating whether or not we are using PyTorch or PyTorch
+            Lightning. This could either be the string “pytorch” or ”pytorch-lightning”.
+        num_processes : int, optional
+            An integer that determines how many different concurrent
+            tasks are allowed. We expect spark.task.gpus = 1 for GPU-enabled training. Default
+            should be 1; we don't want to invoke multiple cores/gpus without explicit mention.
+        local_mode : bool, optional
+            A boolean that determines whether we are using the driver
+            node for training. Default should be false; we don't want to invoke executors without
+            explicit mention.
+        use_gpu : bool, optional
+            A boolean that indicates whether or not we are doing training
+            on the GPU. Note that there are differences in how GPU-enabled code looks like and
+            how CPU-specific code looks like.
+        spark : Optional[SparkSession], optional
+            An optional parameter that allows users to pass in a custom SparkSession argument
+            with a custom conf, by default None
+
+        Raises
+        ------
+        ValueError
+            If any of the parameters are incorrect.
+        """
+        # TODO(SPARK-41915): Remove framework requirement in a future PR.
+        super().__init__(num_processes, local_mode, use_gpu, spark)
+        self.framework = framework
+        self.ssl_conf = "pytorch.spark.distributor.ignoreSsl"  # type: ignore
+        self._validate_input_params()
+
+    def _validate_input_params(self) -> None:
+        """Validates input params
+
+        Raises
+        ------
+        ValueError
+            Thrown when user fails to provide correct input params
+        """
+        super()._validate_input_params()
+        if self.framework not in self.available_frameworks:
+            raise ValueError(
+                f"{self.framework} is not a valid framework."
+                f"Available frameworks: {self.available_frameworks}"
+            )
+
+    def run(self, train_fn: Union[Callable, str], *args: Union[str, int]) -> Union[None, Any]:
+        """Runs distributed training.
+
+        Parameters
+        ----------
+        train_fn : Union[Callable, str]
```

Comment:
per numpydoc (https://numpydoc.readthedocs.io/en/latest/format.html#parameters)

```suggestion
        train_fn : callable object or str
```

Would have to fix all places in this PR.